acceptance-tests/src/test/scala/ch/renku/acceptancetests/tooling/Grammar.scala

```diff
@@ -110,6 +110,15 @@ trait Grammar extends WebElementOps with WebDriverOps with Scripts with Eventual
             s"but gets stuck on $currentUrl for ${(checkInterval * attempt).toSeconds}s"
         }
     }
+
+    @scala.annotation.tailrec
+    def whenUserCanSee(findElement: WebDriver => Option[WebElement], attempt: Int = 1): Unit =
+      if (attempt <= 30 && findElement(webDriver).fold(ifEmpty = false)(_.isDisplayed)) {
+        sleep(1 second)
```

Comment:
Should this be more dynamic or is it reliable enough as a hardcoded sleep?

Reply:
That's always a bit of a problem. It kind of settled down empirically and it's good enough to make the tests pass in all environments. So far I didn't come up with a better dynamic way for this.